src/app/features/ledger/flows/tx-signing/ledger-sign-tx-container.tsx

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useState } from 'react';
+import { Outlet, useLocation, useNavigate } from 'react-router-dom';
+import { Box } from '@stacks/ui';
+import { LedgerError } from '@zondax/ledger-blockstack';
+import get from 'lodash.get';
+
+import { delay, noop } from '@app/common/utils';
+import {
+  getAppVersion,
+  prepareLedgerDeviceConnection,
+  signLedgerTransaction,
+  signTransactionWithSignature,
+  useLedgerResponseState,
+} from '@app/features/ledger/ledger-utils';
+import { RouteUrls } from '@shared/route-urls';
+import { deserializeTransaction } from '@stacks/transactions';
+import { LedgerTxSigningProvider } from '@app/features/ledger/ledger-tx-signing.context';
+import { useCurrentAccount } from '@app/store/accounts/account.hooks';
+import { LoadingKeys } from '@app/common/hooks/use-loading';
+import { useHandleSubmitTransaction } from '@app/common/hooks/use-submit-stx-transaction';
+import { BaseDrawer } from '@app/components/drawer';
+
+import { useLedgerNavigate } from '../../hooks/use-ledger-navigate';
+import { useLedgerAnalytics } from '../../hooks/use-ledger-analytics.hook';
+import { logger } from '@shared/logger';
+
+export function LedgerSignTxContainer() {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const ledgerNavigate = useLedgerNavigate();
+  const ledgerAnalytics = useLedgerAnalytics();
+
+  const account = useCurrentAccount();
+
+  const [unsignedTransaction, setUnsignedTransaction] = useState<null | string>(null);
+
+  useEffect(() => {
+    const tx = get(location.state, 'tx');
+    if (tx) setUnsignedTransaction(tx);
+  }, [location.state]);
+
+  useEffect(() => () => setUnsignedTransaction(null), []);
+
+  const [latestDeviceResponse, setLatestDeviceResponse] = useLedgerResponseState();
+
+  const [awaitingDeviceConnection, setAwaitingDeviceConnection] = useState(false);
+
+  const broadcastTransactionFn = useHandleSubmitTransaction({
+    loadingKey: LoadingKeys.CONFIRM_DRAWER,
+  });
+
+  const signTransaction = async () => {
+    if (!account) return;
+
+    const stacks = await prepareLedgerDeviceConnection({
```

Comment:
It might be more clear to use `stacksApp` or something on these just bc `stacks` is used so often in our codebase?